erpnext/selling/doctype/sales_order/sales_order.py

```diff
@@ -475,6 +484,86 @@ def validate_serial_no_based_delivery(self):
 				Item {0} is added with and without Ensure Delivery by \
 				Serial No.").format(item.item_code))
 
+	def send_delivery_request_to_flowkana(self):
+		"""
+		Ping flowkana with sales order details and map response to an integration request.
+		"""
+		#create line items
+		item_list = []
+		for item in self.items:
+			ivt_id = frappe.get_value("Item", item.item_code, "ivt_id")
+			line_item = {
+				"attributes": {
+				"ivt_id": ivt_id, #change to ivt_id
+				"coa_id": item.batch_no,
+				"external_item_code": item.item_code,
+				"unit_quantity": item.qty,
+				"unit_price": item.rate
+				}
+			}
+			item_list.append(line_item)
+
+		#prepare response json
+		request_json = {
+				"data": {
+					"attributes": {
+						"external_order_id": self.name,
+						"customer_name": self.customer,
+						"customer_license": self.license,
+						"delivery_date": self.delivery_date,
+						"note": "Test Note"
+					},
+					"relationships": {
+						"order_line_items": item_list
+					}
+				}
+			}
+		print("Request JSON: ", request_json)
+		#create integration request
+		integration_request = frappe.new_doc("Integration Request")
+		integration_request.update({
+			"integration_type": "Host",
+			"integration_request_service": "Flowkana",
+			"status": "Queued",
+			"data": json.dumps(request_json, default=json_handler),
+			"reference_doctype": "Sales Order",
+			"reference_docname": self.name
+		})
+		integration_request.insert()
```

Comment:
```suggestion
		integration_request.save(ignore_permissions=True)
```

Reply:
We will need to insert it first

Reply:
- Save inserts it, if the document is not inserted